tooling-cpu-load/src/main/java/org/jboss/eap/qe/microprofile/tooling/cpu/load/utils/ProcessUtils.java

```diff
@@ -0,0 +1,110 @@
+        model.get(ClientConstants.OP_ADDR).add("type", "runtime");
+
+        ModelNodeResult result = client.execute(model);
+        String nodeName = result.get("result").get("name").asString();
+        return Integer.valueOf(nodeName.substring(0, nodeName.indexOf("@")));
+    }
+
+    /**
+     * Returns pid of given process.
+     *
+     * @param process process
+     * @return pid of the process
+     */
+    public static int getProcessId(Process process) {
+
+        int pid;
+        try {
+            Field f = process.getClass().getDeclaredField("pid");
+            f.setAccessible(true);
+            pid = f.getInt(process);
+        } catch (Throwable e) {
+            throw new IllegalStateException("Unable to get process id for " + process.getClass().getSimpleName(), e);
+        }
+
+        if (pid == 0) {
+            throw new IllegalStateException("Process ID is 0 for " + process.getClass().getSimpleName());
+        }
+
+        return pid;
+    }
+
+    /**
+     * Priority can be set -19 to 20, lower number = higher priority
+     *
+     * @param pid id of the process
+     * @param priority priority to be set, -19 to 20, lower number = higher priority
+     */
+    public static void setPriorityToProcess(String pid, int priority) throws Exception {
+
+        if (isPriorityInRange(priority)) {
+
+            if (!System.getProperty("os.name").contains("Linux")) {
+                throw new UnsupportedOperationException(
+                        "Command renice which is used to lower priority of process is supported only on linux. " +
+                                "Current operation system is: " + System.getProperty("os.name").contains("Linux"));
+            }
+
+            Process setPriorityProcess = Runtime.getRuntime().exec("renice -n " + priority + " -p " + pid);
+            if (setPriorityProcess.waitFor() != 0) {
+                throw new Exception("Process: " + setPriorityProcess + " did not exit with code 0.");
+            }
+        } else {
+            throw new Exception("Set process priority between -19 to 20. Priority was set to: " + priority);
+        }
+    }
+
+    private static boolean isPriorityInRange(int priority) {
+        if (priority < -19 || priority > 20) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Binds process with process to given CPU cores.
+     *
+     * @param pid process id
+     * @param cpuIds Specify cores process to bind on - for example "0-2,6" will bind process to 4 CPU cores (0,1,2,6)
+     * @throws Exception throws exception if this operation fails
+     */
+    public static void bindProcessToCPU(int pid, String cpuIds) throws Exception {
+        String cmd;
+        if (System.getProperty("os.name").contains("Linux")
+                && (System.getProperty("os.version").contains("el7") || System.getProperty("os.version").contains("fc2"))) {
```

Comment:
Magic constants

Reply:
thanks, as I've found this check on 3 places I decided to place it in separate method . I've also renamed this class to LinuxProcessUtils as it dependent on Linux environment.